src/isl/isl_constraint.rs

```diff
@@ -103,6 +104,11 @@ impl IslConstraint {
         IslConstraint::CodepointLength(length)
     }
 
+    /// Creates an [IslConstraint::TimestampPrecision] using the range specified in it
+    pub fn timestamp_precision(precision: Range) -> IslConstraint {
```

Comment:
question—Is it possible to change this function signature so that you can't pass in the wrong type of `Range`?

Reply:
I agree that there should be a better representation to `Range` to resolve this issue. I will add a short-term solution that Zack suggested to panic for invalid range types for now.
I will create an issue to keep track of this.

Reply:
Reference issue: https://github.com/amzn/ion-schema-rust/issues/89